cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -313,7 +314,8 @@ private void destroyWorkflow() {
       LOG.error("Failed to store the final workflow token of Workflow {}", workflowRunId, t);
     }
 
-    if (ProgramStatus.COMPLETED != workflowContext.getState().getStatus()) {
+    if (ProgramStatus.COMPLETED != workflowContext.getState().getStatus() ||
+        ("false".equalsIgnoreCase(cConf.get(FieldLineage.FIELD_LINEAGE_EMISSION_ENABLED)))) {
```

Comment:
Any reason not to use io.cdap.cdap.common.conf.Configuration#getBoolean ?

Reply:
It returns NPE when the configuration is not present.